src/raftstore/store/peer.rs

```diff
@@ -873,18 +922,20 @@ impl Peer {
     fn process_raft_cmd(&mut self,
                         index: u64,
                         term: u64,
-                        cmd: RaftCmdRequest)
+                        cmd: RaftCmdRequest,
+                        wb: &mut WriteBatch)
                         -> Result<Option<ExecResult>> {
         if index == 0 {
             return Err(box_err!("processing raft command needs a none zero index"));
         }
 
         let uuid = util::get_uuid_from_req(&cmd).unwrap();
         let cb = self.find_cb(uuid, term, &cmd);
-        let (mut resp, exec_result) = self.apply_raft_cmd(index, term, &cmd).unwrap_or_else(|e| {
-            error!("{} apply raft command err {:?}", self.tag, e);
-            (cmd_resp::new_error(e), None)
-        });
+        let (mut resp, exec_result) = self.apply_raft_cmd(index, term, &cmd, wb)
```

Comment:
Because post_apply will change the content of response to the client, so we can not response to client before post_apply finished. If we don't write the applied result to engine before post_apply execute, post_apply may read stall value, so the order of execution must be 1) collection all to write-batch 2) write to engine 3) post_apply 4) response to client, and I have benched this case, the performance will decreased a lot. Should we use a write-batch for each region, not for all region?